src/storage/adapters/local-storage-adapter.ts

```diff
@@ -0,0 +1,61 @@
+import { promises as fs } from 'node:fs';
+import { dirname, join, normalize } from 'node:path';
+import { type Result, error, ok } from 'core/result';
+import type {
+    StorageAdapter,
+    StorageError,
+    VerificationError,
+} from './storage-adapter';
+
+export class LocalStorageAdapter implements StorageAdapter {
+    private readonly hostingUrl: URL;
+
+    constructor(
+        private readonly storagePath: string,
+        hostingUrl: URL,
+    ) {
+        if (!hostingUrl.href.endsWith('/')) {
+            this.hostingUrl = new URL(`${hostingUrl.href}/`);
+        } else {
+            this.hostingUrl = hostingUrl;
+        }
+    }
+
+    async save(
+        file: File,
+        path: string,
+        // TODO this should return Result<URL>
+    ): Promise<Result<string, StorageError>> {
+        if (path.startsWith('/')) {
+            return this.save(file, path.slice(1));
+        }
```

Comment:
Is this so just so you don't have modify `path` or is there a more elaborate reason for the recursion?

Reply:
Yeah we have a lint rule about not modifying parameters, and I think this is cleaner than
```ts
let actualPath = path;
if (path.startsWith('/')) {
    actualPath = path.slice(1);
}
```